app/routes/ui/scenario.py

```diff
@@ -17,20 +19,35 @@ class UIScenarioHandler(BaseUIHandler):
     async def get(self, scenario_friendly_name, scene_number=0):
         try:
             scene_number_int = int(scene_number)
-            # TODO: caching will be dealt with in a later PR
-            scenario_obj: Scenario = await get_scenario_by_friendly_name_from_postgres(
-                scenario_friendly_name, self.db_session
-            )
+            scenario_dict = await check_and_get_scenario_by_name(scenario_friendly_name)
+            if not scenario_dict:
+                if not self.db_session:
+                    self.db_session = get_session()
+                scenario_obj: Scenario = (
```

Comment:
Never seen this syntax before! Mind explaining quickly what is happening here?

Reply:
Defining `scenario_obj` as type `Scenario`, and the bracket is just something that was added by the linters to make it prettier, to keep the function call on one line!